packages/loader/container-loader/src/retriableDocumentStorageService.ts

```diff
@@ -95,26 +108,32 @@ export class RetriableDocumentStorageService implements IDocumentStorageService,
 			0x251 /* "creation summary has to have seq=0 && handle === undefined" */,
 		);
 		if (context.referenceSequenceNumber !== 0) {
-			return this.internalStorageService.uploadSummaryWithContext(summary, context);
+			return this.internalStorageServiceP.then(async (s) =>
+				s.uploadSummaryWithContext(summary, context),
+			);
 		}
 
 		// Creation flow with attachment blobs - need to do retries!
 		return this.runWithRetry(
-			async () => this.internalStorageService.uploadSummaryWithContext(summary, context),
+			async () =>
+				(this.internalStorageService as IDocumentStorageService).uploadSummaryWithContext(
```

Comment:
we shouldn't be casting here. just use the promise like above

Reply:
thanks, I missed this one.